internal/controllers/apis/configmap/disable.go

```diff
@@ -50,10 +65,15 @@ func MaybeNewDisableStatus(ctx context.Context, client client.Client, disableSou
 		return cm, err
 	}
 
-	isDisabled, reason, err := DisableStatus(getCM, disableSource)
+	result, reason, err := DisableStatus(getCM, disableSource)
 	if err != nil {
 		return nil, err
 	}
+	// we treat pending as disabled
+	// eventually we should probably represent isDisabled by an enum in the API, but for now
```

Comment:
is there some reason not to change from isDisabled -> a state enum now?

i think that by introducing this translation layer, we may be introducing more problems (see my above comment), and that this would be simpler and easier to reason about if DisableStatus() really returned a DisableStatus